pkg/controllers/metrics/policy/controller.go

```diff
@@ -0,0 +1,81 @@
+package policy
+
+import (
+	kyvernov1 "github.com/kyverno/kyverno/api/kyverno/v1"
+	kyvernov1informers "github.com/kyverno/kyverno/pkg/client/informers/externalversions/kyverno/v1"
+	"github.com/kyverno/kyverno/pkg/metrics"
+	controllerutils "github.com/kyverno/kyverno/pkg/utils/controller"
+	kubeutils "github.com/kyverno/kyverno/pkg/utils/kube"
+)
+
+type controller struct {
+	// config
+	metricsConfig *metrics.MetricsConfig
+}
+
+// TODO: this is a very strange controller, it only processes events, this should be changed to a real controller
+// but this is difficult as we currently can't remove existing metrics. To be reviewed when we implement a more
+// solid metrics system.
+func NewController(metricsConfig *metrics.MetricsConfig, cpolInformer kyvernov1informers.ClusterPolicyInformer, polInformer kyvernov1informers.PolicyInformer) {
+	c := controller{
+		metricsConfig: metricsConfig,
+	}
+	controllerutils.AddEventHandlers(cpolInformer.Informer(), c.addPolicy, c.updatePolicy, c.deletePolicy)
+	controllerutils.AddEventHandlers(polInformer.Informer(), c.addNsPolicy, c.updateNsPolicy, c.deleteNsPolicy)
+}
+
+func (c *controller) addPolicy(obj interface{}) {
+	p := obj.(*kyvernov1.ClusterPolicy)
+	// register kyverno_policy_rule_info_total metric concurrently
+	go c.registerPolicyRuleInfoMetricAddPolicy(logger, p)
```

Comment:
do we need to run this in go routines ?

Reply:
i think it should be fine to run in go routines

Reply:
ok 👍 
we need to change the way it works though, that's crazy we don't remove the metrics when a policy is deleted

Reply:
right , ideally we should delete the metrics as well.